src/opt.h

```diff
@@ -20,4 +29,180 @@ namespace ispc {
     corresponds to full optimization.
 */
 void Optimize(llvm::Module *module, int optLevel);
+
+///////////////////////////////////////////////////////////////////////////
+// This is a wrap over class llvm::ModulePassManager. This duplicates PassManager function run()
+//   and adds several add functions with some checks and debug passes.
+//   This wrap can control:
+//   - If we want to switch off optimization with given number.
+//   - If we want to dump LLVM IR after optimization with given number.
+//   - If we want to generate LLVM IR debug for gdb after optimization with given number.
+class DebugModulePassManager {
+  public:
+    DebugModulePassManager(llvm::Module &M, int optLevel) : m_passNumber(0), m_optLevel(optLevel) {
```

Comment:
Could you place the function definitions in the .cpp file? While it might not address a specific issue, but it is a good practice and promotes cleaner coding style.

Reply:
Done